packages/camera/camera_avfoundation/ios/Classes/CameraPlugin.m

```diff
@@ -307,12 +348,40 @@ - (void)createCameraOnSessionQueueWithCreateMethodCall:(FlutterMethodCall *)crea
     if (!strongSelf) return;
 
     NSString *cameraName = createMethodCall.arguments[@"cameraName"];
+
+    NSError *error;
+
+    NSNumber *fps = [CameraPlugin positiveOrNil:createMethodCall.arguments[@"fps"] error:&error];
+    if (error) {
+      [result sendError:error];
```

Comment:
The error message returned from `positiveOrNil` is used for three different variables, so someone getting this exception would have no way of knowing which argument was incorrect; argument errors should be specific. You could make this `positiveNumberValueOrNilForArgument:(NSString*) fromMethod:(FlutterMethodCall*) error:(NSError**)` and then the name of the variable can be used in the errors as well as the lookup.

Reply:
rewitten as `+ (NSNumber *)positiveNumberValueOrNilForArgument:(NSString*)argument fromMethod:(FlutterMethodCall*)flutterMethodCall error:(NSError **)error `